src/region.js

```diff
@@ -42,151 +41,145 @@ var Region = MarionetteObject.extend({
-      // If this happens we need to remove the reference
-      // to the currentView since once a view has been destroyed
-      // we can not reuse it.
-      view.once('destroy', this.empty, this);
-
-      // make this region the view's parent,
-      // It's important that this parent binding happens before rendering
-      // so that any events the child may trigger during render can also be
-      // triggered on the child's ancestor views
-      view._parent = this;
-      this._renderView(view, options);
-
-      if (isChangingView) {
-        this.triggerMethod('before:swapIn', view, this, options);
-      }
-
-      this.triggerMethod('before:show', view, this, options);
-      triggerMethodOn(view, 'before:show', view, this, options);
-
-      // An array of views that we're about to display
-      var attachedRegion = isNodeAttached(this.el);
-
-      // The views that we're about to attach to the document
-      // It's important that we prevent _getNestedViews from being executed unnecessarily
-      // as it's a potentially-slow method
-      var displayedViews = [];
-
-      var attachOptions = _.extend({
-        triggerBeforeAttach: this.triggerBeforeAttach,
-        triggerAttach: this.triggerAttach
-      }, showOptions);
+    if (!shouldShowView) {
+      return this;
+    }
 
-      if (attachedRegion && attachOptions.triggerBeforeAttach) {
-        displayedViews = this._displayedViews(view);
-        this._triggerAttach(displayedViews, 'before:');
-      }
+    this.triggerMethod('before:show', view, this, options);
 
-      this.attachHtml(view, _shouldReplaceElement);
-      this.currentView = view;
+    // We need to listen for if a view is destroyed in a way other than through the region.
+    // If this happens we need to remove the reference to the currentView since once a view
+    // has been destroyed we can not reuse it.
+    view.once('destroy', this.empty, this);
 
-      if (attachedRegion && attachOptions.triggerAttach) {
-        displayedViews = this._displayedViews(view);
-        this._triggerAttach(displayedViews);
-      }
+    // Make this region the view's parent.
+    // It's important that this parent binding happens before rendering so that any events
+    // the child may trigger during render can also be triggered on the child's ancestor views.
+    view._parent = this;
 
-      if (isChangingView) {
-        this.triggerMethod('swapOut', changingView, this, options);
-        this.triggerMethod('swapIn', view, this, options);
-      }
+    // Render the view
+    if (!view.supportsRenderLifecycle) {
+      triggerMethodOn(view, 'before:render', view);
+    }
+    view.render();
+    if (!view.supportsRenderLifecycle) {
+      triggerMethodOn(view, 'render', view);
+    }
 
-      this.triggerMethod('show', view, this, options);
-      triggerMethodOn(view, 'show', view, this, options);
+    // Attach the view
+    if (isAttachedRegion && shouldTriggerAttach) {
```

Comment:
Just an additional though.  I don't know that `preventDestroy` without an `attachView` is very useful if using the view in a region again is going to force a render